pysages/colvars/angles.py

```diff
@@ -121,3 +127,152 @@ def dihedral_angle(p1, p2, p3, p4):
+    Equations 4-14.
+    Notice that for rings with N atoms, there are
+    `int( ( N - 1 )  / 2 - 1 )` phase angles and `int (N / 2 - 1) ` amplitudes.
+    So if the ring contains more than six atoms, there is more than one phase angle;
+    similarly, if the ring contains more than five atoms, there is more than one amplitude.
+    This class (for now) only calculates the first amplitude and the phase angle (`m = 2` in
+    Equations 12 and 13, or see `pysages.colvars.angles.ring_puckering_coordinates` for math).
+    Also, the phase angle obtained via the Cremer-Pople method can be converted
+    to the Altona-Sundaralingam order parameter by adding `pi / 2` to the result
+    and then converting from radians to degrees.
+    Similarly, the amplitude obtained via the Cremer-Pople method can be converted
+    to the Altona-Sundaralingam order parameter (in degrees) by multiplying the result
+    by `1025` degree/nanometer.
+
+    Usage
+    -------
+    cvs = [PhaseAngle([index1, index2, ...])]
+
+    Notice that the phase angle is dependent on the order of the indices. For example,
+    the convention for sugar pucker of ribose in RNA/DNA is: O4', C1', C2', C3', C4'.
+    """
+
+    @property
+    def function(self):
+        return ring_puckering_coordinates
+
+
+def ring_puckering_coordinates(rs):
+    r"""
+    calculate phase angle (first phase angle if N>5) based on Cremer-Pople method.
+    :math:`r0 = 1/N \sum_i^N \vec{r}_i`
+    :math:`\vec{R1} = \sum_i^N (\vec{r}_i -r_c) \sin(2\pi (i-1)/N)`
+    :math:`\vec{R2} = \sum_i^N (\vec{r}_i -r_c) \cos(2\pi (i-1)/N)`
+    :math:`\hat{n} = \vec{R1} \times \vec{R2}/ (|\vec{R1}\times\vec{R2}|)`
+    :math:`z_i = (\vec_{r}_i-r_c) \cdot \hat{n}`
+    :math:`a =  \sqrt(2/N) \sum_i^N z_i \cos(2\pi 2(i-1)/N)`
+    :math:`b = -\sqrt(2/N) \sum_i^N z_i \sin(2\pi 2(i-1)/N)`
+    :math:`q = \sqrt(a^2 + b^2)`
+    :math:`phi = \atan(b / a)`
+
+    Parameters
+    ------------
+    rs: DeviceArray
+        :math: `\vec{r}_i` array of 3D vector in space
+
+    Returns
+    ------------
+    DeviceArray: [q: float, phi: float]
+        :math:`q` in nanometer (if the default length unit for the MD engine is nanometer)
+        :math:`phi` in radians, range -pi to pi.
+    """
+    N = len(rs)
+    r0 = barycenter(rs)
+    rc = rs - r0
+    theta = 2j * np.pi * np.arange(N) / N
+    fourier_coeff = np.exp(theta)
+    R1 = np.dot(rc.T, np.imag(fourier_coeff))
+    # Notice the imag part corresponds to sin. The order of R1/R2 matters
+    # because otherwise the n would be inverted.
+    R2 = np.dot(rc.T, np.real(fourier_coeff))
+    n = np.cross(R1, R2)
+    n /= linalg.norm(n)
+    z = np.dot(rc, n)
+    fourier_coeff2 = np.exp(2 * theta)
+    a = np.sqrt(2 / N) * np.sum(z * np.real(fourier_coeff2))
+    b = -np.sqrt(2 / N) * np.sum(z * np.imag(fourier_coeff2))
+    q = np.sqrt(a**2 + b**2)
+    phi = np.arctan2(b, a)
+    return np.array([q, phi])
```

Comment:
What I had in mind was to return `a` and `b` here and compute `q` and `phi` on the `RingAmplitude` and `RingPhaseAngle`, respectively:
```suggestion
    return np.array([a, b])
```